xgi/classes/simplicialcomplex.py

```diff
@@ -0,0 +1,332 @@
+"""Base class for undirected simplicial complexes.
+
+The SimplicialComplex class allows any hashable object as a node
+and can associate key/value attribute pairs with each undirected simplex and node.
+
+Multi-simplices are not allowed.
+"""
+from copy import deepcopy
+from itertools import combinations
+
+import xgi
+import xgi.convert as convert
+from xgi.classes import Hypergraph
+from xgi.classes.reportviews import (DegreeView, EdgeSizeView, EdgeView,
+                                     NodeView)
+from xgi.exception import XGIError
+from xgi.utils import XGICounter
+
+__all__ = ["SimplicialComplex"]
+
+class SimplicialComplex(Hypergraph) :
+    """A class to represent undirected simplicial complexes."""
+
+    # def is_simplicial_complex(self) :
+    #     """Returns True if all subfaces of each edge exist
+        
+    #     Goes through all simplices, from larger to smaller and stops at first missing subface found.
+    #     """
+        
+    #     # check that all nodes in simplices are in nodes
+    #     nodes_to_have = sorted({node for edge in self.edges for node in edge})
+    #     if not set(nodes_to_have).issubset(self.nodes) :
+    #         print("Some of the nodes members of edges are not in H.nodes.")
+    #         return False
+    #     else : # check that all subfaces of each edge exist
+    #         edges = self.edges
+    #         for edge in edges[::-1] : # loop over edges, from larger to smaller 
+        
+    #             size = len(edge) 
+    #             if size>=3 : # because we already checked nodes 
+    #                 for face in combinations(edge, size-1) : # check if all subfaces are present
+
+    #                     face = tuple(sorted(face))
+    #                     if face not in edges : 
+    #                         print(f"{face} not in SimplicialComplex (first found)")
+    #                         return False
+            
+    #         else : 
+    #             return True 
```

Comment:
Pro-tip: leaving commented out code is usually frowned upon -- that's what version control is for! :)

Reply:
Forgot it there :)